plugins/filter_kubernetes/kube_meta.c

```diff
@@ -226,6 +309,18 @@ static int get_meta_info_from_request(struct flb_kube *ctx,
         return -1;
     }
 
+    /* Refresh HTTP Auth Header if K8s Authorization Token is expired */
```

Comment:
I would like to discuss bit about how we create a new function or not. 
For this case, it will make sense to me to create a new function called 
refresh_kube_token() 
and the logic below could be put into the into that function to do check if we need to refresh or not and how to do refresh. 
In case in the future we need new logic for refresh, we don't need to change this function anymore and more focus on the refresh function.

Reply:
I like Drew's idea. It makes the code easier to read in this function. This logic can encapsulated in another function.